newrelic/resource_newrelic_alert_policy.go

```diff
@@ -73,82 +104,118 @@ func resourceNewRelicAlertPolicyCreate(d *schema.ResourceData, meta interface{})
 	if len(channels) > 0 {
 		channelIDs := expandAlertChannelIDs(channels)
 		matchedChannelIDs, err := findExistingChannelIDs(client, channelIDs)
-
 		if err != nil {
 			return err
 		}
 
 		log.Printf("[INFO] Adding channels %+v to policy %+v", matchedChannelIDs, policy.Name)
 
-		_, err = client.Alerts.UpdatePolicyChannels(policy.ID, matchedChannelIDs)
-
+		_, err = client.Alerts.UpdatePolicyChannels(createResult.ID, matchedChannelIDs)
 		if err != nil {
 			return err
 		}
 	}
 
-	d.SetId(strconv.Itoa(policy.ID))
-
 	return nil
 }
 
 func resourceNewRelicAlertPolicyRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ProviderConfig).NewClient
+	cfg := meta.(*ProviderConfig)
 
-	id, err := strconv.ParseInt(d.Id(), 10, 32)
+	if !cfg.hasNerdGraphCredentials() {
+		return errors.New("err: NerdGraph support not present, but required for Read")
+	}
+
+	client := cfg.NewClient
+
+	ids, err := parseHashedIDs(d.Id())
 	if err != nil {
 		return err
 	}
 
-	log.Printf("[INFO] Reading New Relic alert policy %v", id)
+	var accountID int
+	var policyID int
 
-	policy, err := client.Alerts.GetPolicy(int(id))
+	if len(ids) == 1 {
+		policyID = ids[0]
+		accountID = d.Get("account_id").(int)
 
-	if err != nil {
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
-		}
+	} else if len(ids) == 2 {
+		policyID = ids[0]
+		accountID = ids[1]
```

Comment:
Wondering if you can use the `selectAccountID()` helper function located in `resource_helpers.go` to avoid these conditionals. Since `policyID` should always be `ids[0]`, could use the helper function to grab the account ID.

Reply:
I'll take a look.

Reply:
Done.